--- ./webui/source/Styles/xb3/code/dynamic_dns_edit.php
+++ ./webui/source/Styles/xb3/code/dynamic_dns_edit.php	2015-04-22 14:22:31.206843009 -0600
@@ -157,14 +157,22 @@
 
 			<div class="form-row "  style="float:left">
 				<label for="Service_Provider">Service Provider:</label>
-				<input type="text" value="dyndns.org" id="Service_Provider1"  name="Service_Provider" />
-
+				<!--input type="text" value="dyndns.org" id="Service_Provider1"  name="Service_Provider" /-->
+				<select name="Service_Provider1" id="Service_Provider1">
+				<?php
+					$ids=explode(",",getInstanceIDs("Device.X_CISCO_COM_DDNS.Service."));
+					foreach ($ids as $key=>$j) {
+						$spName		=getStr("Device.X_CISCO_COM_DDNS.Service.".$j.".ServiceName");
+						$serviceStatus	=getStr("Device.X_CISCO_COM_DDNS.Service.".$j.".Enable");
+						if(strcasecmp($sp,$spName) == 0) echo '<option value="'.$spName.'">'.$spName.'</option>';
+						else if(strcasecmp($serviceStatus,"false") == 0) echo '<option value="'.$spName.'">'.$spName.'</option>';
+					}
+				?>
+				</select>
 			</div>
-			<div  style="font-size: .8em;  "></br>Eg:dyndns.org,changeip.com,freedns.afraid.org</div>
 
 		</div>
 
-
 		<div class="form-row odd" >
 			<label for="User_name"> User Name:</label> <input type="text"  id="User_name" name="User_name" class="text" size="35"  value="USG"/>
 		</div>
