--- ./webui/source/Styles/xb3/code/managed_sites.php
+++ ./webui/source/Styles/xb3/code/managed_sites.php	2015-05-04 11:39:36.230783877 -0600
@@ -258,6 +258,12 @@
 			$(btn).click(function(){
 				$('.main_content').hide();
 				$(edit).show();
+			
+	 $(function() {
+$.validator.addMethod("no_space", function(value, element, param) {
+		return !param || /^[a-zA-Z0-9]*$/i.test(value);
+	}, " No spaces. Case sensitive.");
+
 				$(pageFORM).validate({
 					rules: {
 						url: {
@@ -265,7 +271,8 @@
 							url2: true 
 						},
 						Keyword: {
-							required: true
+							 required: true,
+               						 no_space:true
 						},
 						day: {
 							required: function() {
@@ -278,7 +285,7 @@
 					}
 				});
 			});
-
+});
 			//=========================================================
 			var blockTime = "#block-time-" + k;  
 
@@ -440,7 +447,7 @@
     <p class="hidden"><strong>+ADD:</strong> Add a new website or keyword.</p>
     <p class="hidden"><strong>Blocked Sites:</strong> Deny access to specific websites (URLs).</p>
     <p class="hidden"><strong>Blocked Keywords:</strong> Deny access to websites containing specific words.</p>
-    <p class="hidden">The Gateway will block connections to websites on all untrusted computers, based on the specified rules. If you don't want restrictions for a particular computer, select <strong>Yes</strong> under Trusted Computers.</p>
+    <p class="hidden">The Gateway will block connections to websites on all untrusted computers, based on the specified rules. If you don't want restrictions for a particular computer, select <strong>Yes</strong> under <strong>Trusted Computers</strong>.</p>
 	</div>
 
 	<form action="managed_sites.php" method="post"  name="managed_sites">
@@ -639,6 +646,24 @@
 			}
 
 			$Host["$i"]['IPAddress'] = getStr("Device.Hosts.Host." .$hostsInstanceArr["$i"]. ".IPAddress");
+			$IPAddress = $Host["$i"]['IPAddress'];
+			//$IPv4Address	= getStr("Device.Hosts.Host." .$hostsInstanceArr["$i"]. ".IPv4Address.1.IPAddress");
+			$IPv6Address	= getStr("Device.Hosts.Host." .$hostsInstanceArr["$i"]. ".IPv6Address.1.IPAddress");
+			
+			//"Device.Hosts.Host.'$i'.IPv4Address.1.IPAddress" is not updating on GW_IP Change
+			$IPv4Address = $IPAddress;
+
+			//In IPv6 only mode, IPv4=NA
+			if( strpos($IPv4Address, '.') === false ) $IPv4Address = 'NA';
+
+			if (substr($IPv6Address, 0, 5) == "2001:") {
+				$Host["$i"]['IPShow'] = $IPv4Address.'/'.$IPv6Address;
+			}
+			else {
+				//If IPv6 is not global then IPv6=NA
+				$Host["$i"]['IPShow'] = $IPv4Address.'/NA';
+			}
+
 			array_push($HostNameArr, $Host["$i"]['HostName']);
 			array_push($ipAddrArr, $Host["$i"]['IPAddress']);
 			$Host["$i"]['Trusted'] = false;
@@ -679,8 +704,8 @@
 						echo "<tr $odd>
 						<td headers='number' class=\"row-label alt\">" .$k. "</td>
 						<td headers='device-name' id='HostName-" .$k. "' >" .$Host["$i"]['HostName']. "</td>
-						<td headers='IP' id='IPAddress-" .$k. "' >" .$Host["$i"]['IPAddress']. "</td>
-						<td headers='trusted-or-not'>
+						<td headers='IP' id='IPAddress-" .$k. "' >" .$Host["$i"]['IPShow']. "</td>
+						<td headers='trusted-or-not' style='min-width: 92px;'>
 						<span id=\"trusted_user_".$k."\" switch-val=\"".( $Host["$i"]['Trusted'] == 'true' ? "on" : "off" )."\"></span>
 						</td>
 						</tr>
@@ -714,7 +739,7 @@
 		var hostNums = '<?php echo $hostNums; ?>';
 		var hostNameArr = <?php echo json_encode($HostNameArr); ?>;
 		var ipAddrArr = <?php echo json_encode($ipAddrArr); ?>;
-
+//console.log(ipAddrArr);
 	</script>
 
 <?php 
@@ -811,13 +836,13 @@
         			<a rel=\"weekday\" href=\"#select_all\"  class=\"weekday_select_all\">Select All</a> | <a rel=\"weekday\"  href=\"#select_none\" class=\"weekday_select_none\">Select None</a>
                 </div>
         		<div class=\"form-row weekday\">
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"monday-" .$ID. "\"  value=\"Mon\"" . ( (stristr($block_days, "Mon") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). "/><label class=\"checkbox\" for=\"monday-" .$ID. "\">Monday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"tuesday-" .$ID. "\"  value=\"Tue\"" . ( (stristr($block_days, "Tue") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"tuesday-" .$ID. "\">Tuesday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"wednesday-" .$ID. "\"  value=\"Wed\"" . ((stristr($block_days, "Wed") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"wednesday-" .$ID. "\">Wednesday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"thursday-" .$ID. "\"  value=\"Thu\"" . ( (stristr($block_days, "Thu") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"thursday-" .$ID. "\">Thursday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"friday-" .$ID. "\"  value=\"Fri\"" . ( (stristr($block_days, "Fri") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"friday-" .$ID. "\">Friday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"saturday-" .$ID. "\"  value=\"Sat\"" . ((stristr($block_days, "Sat") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"saturday-" .$ID. "\">Saturday</label><br />
-        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"sunday-" .$ID. "\"  value=\"Sun\"" . ( (stristr($block_days, "Sun") != false || $blockedInfo['AlwaysBlock'] == 'true') ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"sunday-" .$ID. "\">Sunday</label>
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"monday-" .$ID. "\"  value=\"Mon\"" . ( (stristr($block_days, "Mon") != false) ? " checked='checked'" : "" ). "/><label class=\"checkbox\" for=\"monday-" .$ID. "\">Monday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"tuesday-" .$ID. "\"  value=\"Tue\"" . ( (stristr($block_days, "Tue") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"tuesday-" .$ID. "\">Tuesday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"wednesday-" .$ID. "\"  value=\"Wed\"" . ((stristr($block_days, "Wed") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"wednesday-" .$ID. "\">Wednesday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"thursday-" .$ID. "\"  value=\"Thu\"" . ( (stristr($block_days, "Thu") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"thursday-" .$ID. "\">Thursday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"friday-" .$ID. "\"  value=\"Fri\"" . ( (stristr($block_days, "Fri") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"friday-" .$ID. "\">Friday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"saturday-" .$ID. "\"  value=\"Sat\"" . ((stristr($block_days, "Sat") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"saturday-" .$ID. "\">Saturday</label><br />
+        			<input  name=\"day\" class=\"blockedDay-" .$ID. "\"  type=\"checkbox\" id=\"sunday-" .$ID. "\"  value=\"Sun\"" . ( (stristr($block_days, "Sun") != false) ? " checked='checked'" : "" ). " /><label class=\"checkbox\" for=\"sunday-" .$ID. "\">Sunday</label>
         		</div>
         	</div> <!-- end #block-time -->
 
