--- ./webui/source/Styles/xb3/code/managed_devices_edit_allowed.php
+++ ./webui/source/Styles/xb3/code/managed_devices_edit_allowed.php	2015-04-22 14:22:31.206843009 -0600
@@ -255,10 +255,10 @@
 					});
 				} 
 			} else {
-				alert("MAC is not valid! Can not be saved.");
+				jAlert("MAC is not valid! Can not be saved.");
 			}
 		} else {
-				alert("Not valid! Can not be saved.");
+				jAlert("Not valid! Can not be saved.");
 		}
 	});
 	
@@ -266,7 +266,7 @@
 </script>
 
 <div id="content">
-	<h1>Parental Control > Manage Devices > Edit Allowed Device</h1>
+	<h1>Parental Control > Managed Devices > Edit Allowed Device</h1>
 	<form id="pageForm"  method="post"> 
 
 	<div class="module">
