--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_wizard_step1.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_wizard_step1.php	2015-04-22 14:22:31.198839009 -0600
@@ -4,6 +4,9 @@
 $jsConfig = $_REQUEST['configInfo'];
 //$jsConfig = '{"newPassword": "11111111", "instanceNum": "1", "oldPassword": "111"}';
 
+//if request is from "password_change.php"
+//$jsConfig = '{"newPassword": "11111111", "instanceNum": "1", "oldPassword": "111", "ChangePassword": "true"}';
+
 $arConfig = json_decode($jsConfig, true);
 //print_r($arConfig);
 
@@ -13,8 +16,11 @@
 
 if (getStr("Device.Users.User.$i.X_CISCO_COM_Password") ==  $arConfig['oldPassword']) 
 {
+	if($arConfig['ChangePassword']){
+		setStr("Device.Users.User.3.X_CISCO_COM_Password", $arConfig['newPassword'], true);
+	}
 	$p_status = "Match";
-	setStr("Device.Users.User.$i.X_CISCO_COM_Password", $arConfig['newPassword'], true);	
+	//setStr("Device.Users.User.$i.X_CISCO_COM_Password", $arConfig['newPassword'], true);	
 }
 
 $arConfig = array('p_status'=>$p_status);
