--- ./webui/source/Styles/xb3/code/actionHandler/ajax_hs_port_forwarding.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajax_hs_port_forwarding.php	2015-05-04 11:39:36.230783877 -0600
@@ -62,21 +62,42 @@
 		//$result="";
 		$ids=explode(",",getInstanceIDs("Device.NAT.PortMapping."));
 		foreach ($ids as $key=>$j) {
-			if (getStr("Device.NAT.PortMapping.".$j.".LeaseDuration")==0 && getStr("Device.NAT.PortMapping.".$j.".InternalPort")!=0){
+			if (getStr("Device.NAT.PortMapping.".$j.".LeaseDuration")==0){
 				$arrayName=getStr("Device.NAT.PortMapping.".$j.".Description");
 				$arrayIP=getStr("Device.NAT.PortMapping.".$j.".InternalClient");
 				$arrayType=getStr("Device.NAT.PortMapping.".$j.".Protocol");
 				$arraySPort=getStr("Device.NAT.PortMapping.".$j.".ExternalPort");
 				$arrayEPort=getStr("Device.NAT.PortMapping.".$j.".ExternalPortEndRange");
-				$arrayPPort=getStr("Device.NAT.PortMapping.".$j.".InternalPort");
+				$InternalPort = getStr("Device.NAT.PortMapping.$j.InternalPort");
 				if($name==$arrayName) { 
-					$result.="Service name has been used!\n";
+					if($InternalPort !=0){
+						$result.="Service name has been used in HS Port Forwarding service!\n";
 					break;
+					} else {
+						$result.="Service name has been used in Port Forwarding service!\n";
+						break;
+					}
 				} else if($type=="BOTH"||$arrayType=="BOTH"||$type==$arrayType){
+					if($arrayIP==$ip && $InternalPort==$priport){
+						if($InternalPort !=0){
+							$result.="Conflict with other HS Port Forwarding service. Please check Private Port(s) and IP!";
+							break;
+						} else {
+							$result.="Conflict with other Port Forwarding service. Please check port and IP!";
+							break;
+						}
+					}
+					else if($arrayIP==$ip) {
 					$porttest=PORTTEST($startport,$endport,$arraySPort,$arrayEPort);
 					if ($porttest==1) {
-						$result.="Conflict with other service. Please check port and IP!";
+							if($InternalPort !=0){
+								$result.="Conflict with other HS Port Forwarding service. Please check Public port and IP!";
 						break;
+							} else {
+								$result.="Conflict with other Port Forwarding service. Please check Start/End port and IP!";
+								break;
+							}
+						}
 					}
 				}
 			}
@@ -104,13 +125,100 @@
 	}
 }
 
+
+if (isset($_POST['edit'])){
+//data:{edit:"true",id:id,name:name,type:type,ip:ip,startport:startport,endport:endport,priport:priport,enportrange:enportrange},
+	$i=$_POST['id'];
+	$name=$_POST['name'];
+	$type=$_POST['type'];
+	if ($type=="TCP/UDP") $type="BOTH";
+	$ip=$_POST['ip'];
+	$sport=$_POST['startport'];
+	$eport=$_POST['endport'];
+	$priport=$_POST['priport'];
+	$enportrange=$_POST['enportrange'];
+	// $result="";
+	$ids=explode(",",getInstanceIDs("Device.NAT.PortMapping."));
+	foreach ($ids as $key=>$j) {
+		if ($i==$j) continue;
+		if (getStr("Device.NAT.PortMapping.".$j.".LeaseDuration")==0){
+			$arrayName  = getStr("Device.NAT.PortMapping.".$j.".Description");
+			$arrayIP    = getStr("Device.NAT.PortMapping.".$j.".InternalClient");
+			$arrayType  = getStr("Device.NAT.PortMapping.".$j.".Protocol");
+			$arraySPort = getStr("Device.NAT.PortMapping.".$j.".ExternalPort");
+			$arrayEPort = getStr("Device.NAT.PortMapping.".$j.".ExternalPortEndRange");
+			$InternalPort = getStr("Device.NAT.PortMapping.$j.InternalPort");
+			if($name==$arrayName) { 
+				if($InternalPort !=0){
+					$result.="Service name has been used in HS Port Forwarding service!\n";
+					break;
+				} else {
+					$result.="Service name has been used in Port Forwarding service!\n";
+					break;
+				}
+			}
+			else if($type=="BOTH"||$arrayType=="BOTH"||$type==$arrayType){
+				if($arrayIP==$ip && $InternalPort==$priport){
+					if($InternalPort !=0){
+						$result.="Conflict with other HS Port Forwarding service. Please check Private Port(s) and IP!";
+						break;
+					} else {
+						$result.="Conflict with other Port Forwarding service. Please check port and IP!";
+						break;
+					}
+				}
+				else if($arrayIP==$ip) {
+					$porttest=PORTTEST($startport,$endport,$arraySPort,$arrayEPort);
+					if ($porttest==1) {
+						if($InternalPort !=0){
+							$result.="Conflict with other HS Port Forwarding service. Please check Public port and IP!";
+							break;
+						} else {
+							$result.="Conflict with other Port Forwarding service. Please check Start/End port and IP!";
+							break;
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if ($result=="") {
+		$rootObjName ="Device.NAT.PortMapping.";
+		$paramArray = 
+			array (
+				array("Device.NAT.PortMapping.".$i.".Enable", "bool", $enportrange),
+				array("Device.NAT.PortMapping.".$i.".InternalClient", "string", $ip),
+				array("Device.NAT.PortMapping.".$i.".InternalPort", "uint", $priport),
+				array("Device.NAT.PortMapping.".$i.".ExternalPort", "uint", $sport),
+				array("Device.NAT.PortMapping.".$i.".ExternalPortEndRange", "uint", $eport),
+				array("Device.NAT.PortMapping.".$i.".Protocol", "string", $type),
+				array("Device.NAT.PortMapping.".$i.".Description", "string", $name),
+			);
+		$retStatus = DmExtSetStrsWithRootObj($rootObjName, TRUE, $paramArray);	
+		if (!$retStatus){$result="Success!";}
+	}
+}
+
 if (isset($_POST['active'])){
 	//this is to enable/disable PortActive
 	$isChecked=$_POST['isChecked'];
 	$i=$_POST['id'];
-	setStr("Device.NAT.PortMapping.".$i.".Enable",$isChecked,true);
+	if (setStr("Device.NAT.PortMapping.$i.Enable",$isChecked,true) === true) {
+		
+		$result="Success!";
+	}
 }
 
+if (isset($_REQUEST['del'])){
+	delTblObj("Device.NAT.PortMapping.".$_REQUEST['del'].".");
+	
+	Header("Location:../hs_port_forwarding.php");
+	
+	exit;
+}
+
+
 if ($result=="") {
 //the set operation failure due to conflict with port trigger rules or ...
 //so need to remove the '0.0.0.0' entry
