--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_hardware_lan.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_hardware_lan.php	2015-04-22 14:22:31.198839009 -0600
@@ -38,8 +38,11 @@
 		}
 	}
 	else {
-		if (setStr($pLanBridgeHSPortEnablePath, "true", true) !== true
-			|| setStr($HSBridgePortEnablePath, "false", true) !== true) {
+		#zqiu: port should be removed first and then added to another vlan
+		#if (setStr($pLanBridgeHSPortEnablePath, "true", true) !== true
+		#	|| setStr($HSBridgePortEnablePath, "false", true) !== true) {
+		if (setStr($HSBridgePortEnablePath, "false", true) !== true
+                        || setStr($pLanBridgeHSPortEnablePath, "true", true) !== true) {
 			throw new Exception('failed to set parameters to backend');
 		}
 	}
