--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_Reset_Restore.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_Reset_Restore.php	2015-05-04 11:39:36.230783877 -0600
@@ -11,6 +11,25 @@
 
 $ret = array();
 
+//>>zqiu
+function delMacFilterTable( $ssid_list ) {
+	$ssids = explode(" ", $ssid_list);
+	foreach ($ssids as $i)	{
+		$old_id = array_filter(explode(",",getInstanceIds("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MacFilterTable.")));
+		foreach ($old_id as $j) {
+			delTblObj("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MacFilterTable.$j.");
+		}
+		setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.Enable", false, true);
+		setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.FilterAsBlackList", false, true);
+	}
+}
+
+function delMacFilterTables(  ) {
+	delMacFilterTable("1 2 3 5 6");
+	//For WECB
+	setStr("Device.MoCA.X_CISCO_COM_WiFi_Extender.X_CISCO_COM_SSID_Updated", "true", true);
+}
+//<<
 switch ($infoArray[0]) {
 	case "btn1" :
 		$ret["reboot"] = true;
@@ -34,9 +53,26 @@
 	case "btn4" :
 		$ret["wifi"] = true;
 		echo json_encode($ret);	
-		setStr("Device.X_CISCO_COM_DeviceControl.FactoryReset", $infoArray[1],true);
+		delMacFilterTable("1 2");
+		//setStr("Device.X_CISCO_COM_DeviceControl.FactoryReset", $infoArray[1],true);
 		//when restore, radio can be restart, but also need to force it when no change
-		setStr("Device.WiFi.X_CISCO_COM_ResetRadios", "true", true);
+		//setStr("Device.WiFi.X_CISCO_COM_ResetRadios", "true", true);
+		setStr("Device.WiFi.X_CISCO_COM_FactoryResetRadioAndAp", "1,2;1,2",true);	//radio 1, radio 2; Ap 1, Ap 2
+		//For WECB
+		setStr("Device.MoCA.X_CISCO_COM_WiFi_Extender.X_CISCO_COM_SSID_Updated", "true", true);
+		exit(0);
+	case "FactoryResetRadioAndAp" :
+		$ret["wifi"] = true;
+		echo json_encode($ret);
+		$idxArr = explode(";", $infoArray[1]);
+		//$radioIndex=$idxArr[0];
+		$apIndex=$idxArr[1];		
+		if($apIndex != "0") {
+			delMacFilterTable( "$apIndex" );
+			//For WECB
+			setStr("Device.MoCA.X_CISCO_COM_WiFi_Extender.X_CISCO_COM_SSID_Updated", "true", true);
+		}		
+		setStr("Device.WiFi.X_CISCO_COM_FactoryResetRadioAndAp", $infoArray[1],true);		
 		exit(0);
 	case "btn5" :
 		$ret["reboot"] = true;
