--- ./webui/source/Styles/xb3/code/managed_devices_add_computer_blocked.php
+++ ./webui/source/Styles/xb3/code/managed_devices_add_computer_blocked.php	2015-04-22 14:22:31.206843009 -0600
@@ -10,7 +10,7 @@
 
 <script type="text/javascript">
 $(document).ready(function() {
-    comcast.page.init("Parental Control > Manage Devices > Add Blocked Device", "nav-devices");
+    comcast.page.init("Parental Control > Managed Devices > Add Blocked Device", "nav-devices");
     $("input[name='computer']").focus();
 
 /*	$("input[name='computer']").click(function(event,value) {
@@ -217,10 +217,10 @@
 					});
 				} 
 			} else {
-				alert("MAC is not valid! Can not be saved.");
+				jAlert("MAC is not valid! Can not be saved.");
 			}
 		} else {
-				alert("Not valid! Can not be saved.");
+				jAlert("Not valid! Can not be saved.");
 		}
 	});
 	
@@ -228,7 +228,7 @@
 </script>
 
 <div id="content">
-	<h1>Parental Control > Manage Devices > Add Blocked Device</h1>
+	<h1>Parental Control > Managed Devices > Add Blocked Device</h1>
 	<form id="pageForm" action="managed_devices.php" method="post">
 
 	<div class="module">
@@ -254,6 +254,7 @@
 						if ($iclass=="") {$iclass="odd";} else {$iclass="";}
 						$hostName = getStr("Device.Hosts.Host.".$i.".HostName");
 						$hostMac = getStr("Device.Hosts.Host.".$i.".PhysAddress");
+						
 						echo "
 							<tr class=$iclass>
 								<th class=\"row-label alt\"><input name=\"computer\" id=\"$hostMac\" type=\"radio\" value=\"$hostName\" /></th>
