--- ./webui/source/Styles/xb3/code/at_a_glance.php
+++ ./webui/source/Styles/xb3/code/at_a_glance.php	2015-05-04 11:39:36.230783877 -0600
@@ -35,12 +35,12 @@
 	
 	function changeBridge(isBridgeModelEnable) {
 		var cnt = 90;
-		jProgress('Waiting for fully applied in <b id="cnt">' + cnt + '</b> seconds, please be patient...', 600);
+		jProgress('Bridge Mode changes will be fully applied in <b id="cnt">' + cnt + '</b> seconds, please be patient...', 600);
 		$.ajax({
 		type:"POST",
 		url:"actionHandler/ajax_at_a_glance.php",
 		data:{Bridge:"true",isBridgeModel:isBridgeModelEnable},
-		success:function(result){
+		success:function(){
 				// don't do jHide, with latest firmware, switch bridge will return before httpd restart, GUI reload at return will cause 500 error. So wait until http restored. 
 				var hCnt = setInterval(function(){
 					$("#cnt").text(cnt--);
@@ -56,7 +56,7 @@
 		if ('Enabled' == isBridgeModelEnable) {
 			setTimeout(function(){
 				jConfirm(
-				"Enabling Bridge Mode will disable Router functionality of Gateway and turn off the private Wi-Fi network. Are you sure you want to Continue?"
+				"Enabling Bridge Mode will disable Router functionality of gateway and turn off the private Wi-Fi network. Are you sure you want to continue?"
 				,"WARNING:"
 				,function(ret) {
 					if(ret) {
@@ -111,6 +111,87 @@
 		<p class="hidden"> Select <strong>VIEW CONNECTED DEVICES </strong>to manage devices connected to your network.</p>
 	</div>
 	
+
+	<?php
+	//Home Network WiFi Settings
+	//Only do WiFi SSID check if we are not in power saving mode
+	if ("Disabled"==$_SESSION["psmMode"]) {
+		//Assumes that private network is always SSID's 1 and 2
+		$wifi_24_enabled = getStr("Device.WiFi.SSID.1.Enable");
+		$wifi_24_ssid = getStr("Device.WiFi.SSID.1.SSID");
+		$wifi_24_passkey = getStr("Device.WiFi.AccessPoint.1.Security.X_CISCO_COM_KeyPassphrase");
+		$wifi_50_enabled = getStr("Device.WiFi.SSID.2.Enable");
+		$wifi_50_ssid = getStr("Device.WiFi.SSID.2.SSID");
+		$wifi_50_passkey = getStr("Device.WiFi.AccessPoint.2.Security.X_CISCO_COM_KeyPassphrase");
+		//If at least one private SSID is enabled
+		if ( $bridge_mode == "router" && ("true" == $wifi_24_enabled || "true" == $wifi_50_enabled) ) {
+			echo '<div class="module forms" id="wifi-config">';
+				echo '<div>';
+					echo '<h2>Wi-Fi Configuration</h2>';
+				echo '</div>';
+
+				//If both 2.4ghz and 5ghz ssid's and passkeys are the same, or only one is active, then just show one row
+				if ((($wifi_24_ssid == $wifi_50_ssid) && ($wifi_24_passkey == $wifi_50_passkey)) || !("true" == $wifi_24_enabled && "true" == $wifi_50_enabled)) {
+					//Figure out whice one is active
+					if ("true" == $wifi_24_enabled) {
+						$wifi_ssid = $wifi_24_ssid;
+						$wifi_passkey = $wifi_24_passkey;
+					} else {
+						$wifi_ssid = $wifi_50_ssid;
+						$wifi_passkey = $wifi_50_passkey;
+					}
+			
+				echo '<div class="form-row even">';
+					echo '<div class="form-row even">';
+						echo '<span class="readonlyLabel">Wi-Fi SSID:</span>';
+						echo '<span class="value">';echo $wifi_ssid;echo '</span>';
+					echo '</div>';
+					if ("admin" == $_SESSION["loginuser"]) {
+						echo '<div class="form-row even">';
+							echo '<span class="readonlyLabel">Wi-Fi Passkey:</span>';
+							echo '<span class="value">';echo $wifi_passkey;echo '</span>';
+						echo '</div>';
+					}
+				echo '</div>';
+
+				} else {
+				echo '<div class="form-row even">';
+					echo '<div class="form-row even">';
+						echo '<span class="readonlyLabel">Wi-Fi SSID (2.4Ghz):</span>';
+						echo '<span class="value">';echo $wifi_24_ssid;echo '</span>';
+					echo '</div>';
+					if ("admin" == $_SESSION["loginuser"]) {
+						echo '<div class="form-row even">';
+							echo '<span class="readonlyLabel">Wi-Fi Passkey (2.4Ghz):</span>';
+							echo '<span class="value">';echo $wifi_24_passkey;echo '</span>';
+						echo '</div>';
+					}
+				echo '</div>';
+				echo '<div class="form-row odd">';
+					echo '<div class="form-row odd">';
+						echo '<span class="readonlyLabel">Wi-Fi SSID (5Ghz):</span>';
+						echo '<span class="value">';echo $wifi_50_ssid;echo '</span>';
+					echo '</div>';
+					if ("admin" == $_SESSION["loginuser"]) { 
+						echo '<div class="form-row odd">';
+							echo '<span class="readonlyLabel">Wi-Fi Passkey (5Ghz):</span>';
+							echo '<span class="value">';echo $wifi_50_passkey;echo '</span>';
+						echo '</div>';
+					}
+				echo '</div>';
+			}
+			echo '</div>';
+		}
+	} else {
+	//Power Saving Mode is Enabled
+		echo '<div class="module psm">';
+			echo '<div class="select-row">';
+				echo '<span class="readonlyLabel label">Power Saving Mode is enabled!</span>';
+			echo '</div>';
+		echo '</div>';
+	}
+	?>
+	
 	<div class="module div-bridge">
 		<div class="select-row">
 			<span class="readonlyLabel label">Bridge Mode:</span>
@@ -118,6 +199,7 @@
 		</div>
 	</div>
 
+
 	<div class="module forms">
 		<input type="button" onClick="javascript:popUp('at_downloading.php')" class="btn" value="Save Current Configuration"/>
 		<input type="button" onClick="javascript:popUp('at_mycomputer.php')" class="btn" value="Restore Saved Configuration"/>
