--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_edit_blockedSite.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_edit_blockedSite.php	2015-04-22 14:22:31.198839009 -0600
@@ -1,13 +1,59 @@
 <?php 
 
+function time_date_conflict($TD1, $TD2) {
+	$ret = false;
+	$days1 = explode(",", $TD1[2]);
+	$days2 = explode(",", $TD2[2]);
+
+	foreach ($days1 as &$value) {
+		if (in_array($value, $days2)) {
+			//deMorgan's law - to find if ranges are overlapping
+			//(StartA <= EndB)  and  (EndA >= StartB)
+			if((strtotime($TD1[0]) < strtotime($TD2[1])) and (strtotime($TD1[1]) > strtotime($TD2[0]))){
+	  			$ret = true;
+	  			break;
+			} 
+		}
+	}
+	return $ret;
+}
+
 $blockedSiteInfo = json_decode($_REQUEST['BlockInfo'], true);
 
 $objPrefix = "Device.X_Comcast_com_ParentalControl.ManagedSites.BlockedSite.";
 $rootObjName = $objPrefix;
 $index = $blockedSiteInfo['InstanceID'];
 
+$idArr = explode(",", getInstanceIds("Device.X_Comcast_com_ParentalControl.ManagedSites.BlockedSite."));
+
+$block=$blockedSiteInfo['alwaysBlock'];
+$startTime=$blockedSiteInfo['StartTime'];
+$endTime=$blockedSiteInfo['EndTime'];
+$blockDays=$blockedSiteInfo['blockedDays'];
+
+$result = "";
+
 if( array_key_exists('URL', $blockedSiteInfo) ) {
 	//this is to edit blocked URL
+	//firstly, check whether URL exist or not
+	$url = $blockedSiteInfo['URL'];
+    	foreach ($idArr as $key => $value) {
+		if ($index==$value) continue;
+		$always_Block = getStr($objPrefix.$value.".AlwaysBlock");
+		$start_Time = getStr($objPrefix.$value.".StartTime");
+		$end_Time = getStr($objPrefix.$value.".EndTime");
+		$block_Days = getStr($objPrefix.$value.".BlockDays");
+
+		//Check for time and day conflicts
+		$TD1=array($startTime, $endTime, $blockDays);
+		$TD2=array($start_Time, $end_Time, $block_Days);
+		if (($url == getStr($objPrefix.$value.".Site")) && ((($always_Block == "true") || ($block == "true") || time_date_conflict($TD1, $TD2)))){
+			$result .= "Conflict with other blocked site rule. Please check your input!";
+			break;
+		}
+	}
+
+	if ($result == ""){
 	if ($blockedSiteInfo['alwaysBlock'] == "true"){
 		$paramArray = 
 			array (
@@ -54,8 +100,27 @@
 */	
 	}
 }
+}
 else{
 	//this is to edit blocked Keyword
+	$keyword = $blockedSiteInfo['Keyword'];
+    	foreach ($idArr as $key => $value) {
+		if ($index==$value) continue;
+	    	$always_Block = getStr($objPrefix.$value.".AlwaysBlock");
+		$start_Time = getStr($objPrefix.$value.".StartTime");
+		$end_Time = getStr($objPrefix.$value.".EndTime");
+		$block_Days = getStr($objPrefix.$value.".BlockDays");
+
+		//Check for time and day conflicts
+		$TD1=array($startTime, $endTime, $blockDays);
+		$TD2=array($start_Time, $end_Time, $block_Days);
+		if (($keyword == getStr($objPrefix.$value.".Site")) && ((($always_Block == "true") || ($block == "true") || time_date_conflict($TD1, $TD2)))){
+			$result .= "Conflict with other blocked Keyword rule. Please check your input!";
+			break;
+		}
+	}
+
+	if ($result == ""){
 	if ($blockedSiteInfo['alwaysBlock'] == "true"){
 		$paramArray = 
 			array (
@@ -103,5 +168,8 @@
 */	
 	}
 }
+}
+
+echo $result;
 
 ?>
