--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_firewall_config.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_firewall_config.php	2015-04-22 14:22:31.198839009 -0600
@@ -8,50 +8,51 @@
 {
     if ( $firewall_config['block_http'] == "Enabled" )
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTP", "true", false);
-		setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTPs", "true", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTP", "true", true);
+		setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTPs", "true", true);
     }
     else
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTP", "false", false);
-		setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTPs", "false", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTP", "false", true);
+		setStr("Device.X_CISCO_COM_Security.Firewall.FilterHTTPs", "false", true);
     }    
-
+	//sleep(1);
     if ( $firewall_config['block_icmp'] == "Enabled" )
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterAnonymousInternetRequests", "true", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterAnonymousInternetRequests", "true", true);
     }
     else
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterAnonymousInternetRequests", "false", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterAnonymousInternetRequests", "false", true);
     }
-    
+    //sleep(1);
     if ( $firewall_config['block_multicast'] == "Enabled" )
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterMulticast", "true", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterMulticast", "true", true);
     }
     else
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterMulticast", "false", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterMulticast", "false", true);
     }
-    
+    //sleep(1);
     if ( $firewall_config['block_peer'] == "Enabled" )
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterP2P", "true", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterP2P", "true", true);
     }
     else
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterP2P", "false", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterP2P", "false", true);
     }
-    
+    //sleep(1);
     if ( $firewall_config['block_ident'] == "Enabled" )
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterIdent", "true", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterIdent", "true", true);
     }
     else
     {
-        setStr("Device.X_CISCO_COM_Security.Firewall.FilterIdent", "false", false);
+        setStr("Device.X_CISCO_COM_Security.Firewall.FilterIdent", "false", true);
     }    
+	//sleep(1);   
 }
 
 setStr("Device.X_CISCO_COM_Security.Firewall.FirewallLevel", $firewall_config['firewallLevel'], true);
