--- ./webui/source/Styles/xb3/code/managed_services_add.php
+++ ./webui/source/Styles/xb3/code/managed_services_add.php	2015-04-22 14:22:31.206843009 -0600
@@ -56,7 +56,7 @@
 	
 	$("#pageForm").validate({
 	   rules: {
-	       service: {
+	       user_defined_service: {
 	           required: true
 	       }
 	       ,start_port: {
@@ -190,7 +190,7 @@
 			<h2>Add Service to be Blocked</h2>
 
             <div class="form-row">
-				<label for="service">User Defined Service:</label>
+				<label for="user_defined_service">User Defined Service:</label>
 				<input type="text" id="user_defined_service" value="" name="user_defined_service" class="text" />
 			</div>
 
