--- ./webui/source/Styles/xb3/code/port_forwarding_edit.php
+++ ./webui/source/Styles/xb3/code/port_forwarding_edit.php	2015-04-22 14:22:31.206843009 -0600
@@ -10,13 +10,24 @@
 
 <?php 
 //add by yaosheng 
-$LanSubnetMask = getStr("Device.X_CISCO_COM_DeviceControl.LanManagementEntry.1.LanSubnetMask");
 $LanGwIP = getStr("Device.X_CISCO_COM_DeviceControl.LanManagementEntry.1.LanIPAddress");
+$LanSubnetMask 	= getStr("Device.X_CISCO_COM_DeviceControl.LanManagementEntry.1.LanSubnetMask");
+$beginAddr 	= getStr("Device.DHCPv4.Server.Pool.1.MinAddress");
+$endAddr 	= getStr("Device.DHCPv4.Server.Pool.1.MaxAddress");
 
 $i = $_GET['id'];
 $service_name = getStr("Device.NAT.PortMapping.$i.Description");
 $v6ServerIP = getStr("Device.NAT.PortMapping.$i.X_CISCO_COM_InternalClientV6");
+$startport = getStr("Device.NAT.PortMapping.".$i.".ExternalPort"); 
+$endport   = getStr("Device.NAT.PortMapping.".$i.".ExternalPortEndRange");
 
+$DeviceMode = getStr("Device.X_CISCO_COM_DeviceControl.DeviceMode");
+//$DeviceMode = "IPv6";
+$state = getStr("Device.DHCPv6.Server.X_CISCO_COM_Type");
+//2040::/64, 2040:1::/64, 2040:1:2::/64 and 2040:1:2:3::/64
+$prefix_arr = explode('::/', getStr("Device.IP.Interface.1.IPv6Prefix.1.Prefix"));
+$v6_begin_addr = getStr("Device.DHCPv6.Server.Pool.1.PrefixRangeBegin");
+$v6_end_addr = getStr("Device.DHCPv6.Server.Pool.1.PrefixRangeEnd");
 ?>
 
 <style type="text/css">
@@ -38,10 +49,16 @@
     $('#service_name').focus();
 
     var jsNetMask = "<?php echo $LanSubnetMask; ?>";
-    //alert(typeof(jsNetMask));
+    var beginAddr	= "<?php echo $beginAddr; ?>";
+    var endAddr		= "<?php echo $endAddr; ?>";
+    var beginArr	= beginAddr.split(".");
+    var endArr		= endAddr.split(".");
+
     var jsGwIP = "<?php echo $LanGwIP; ?>".split(".");
     var jsGatewayIP = "<?php echo $LanGwIP; ?>";
     var jsV6ServerIP = "<?php echo $v6ServerIP; ?>";
+    var DeviceMode = "<?php echo $DeviceMode; ?>";
+
 
 function populateIPv6Addr(v6addr){
 
@@ -74,12 +91,15 @@
 }
 
 function IsBlank(id_prefix){
+	//Don't check for - ip6_address_r[1-4]
 	var ret = true;
 	$('[id^="'+id_prefix+'"]').each(function(){
+		if($(this).attr('id').search(/^ip6_address_r[1-4]$/) == "-1"){
 		if ($(this).val().replace(/\s/g, '') != ""){
 			ret = false;
 			return false;
 		}
+		}
 	});
 	return ret;
 }
@@ -107,6 +127,9 @@
 	jQuery.validator.addMethod("ip4",function(value,element){
 		return this.optional(element) || (value.match(/^\d+$/g) && value >= 1 && value <= 254);
 	}, "Please enter a valid IP address.");
+	jQuery.validator.addMethod("ip4_end",function(value,element){
+		return this.optional(element) || (value.match(/^\d+$/g) && value >= 1 && value <= 253);
+	}, "Please enter a valid IP address.");
 	jQuery.validator.addMethod("port",function(value,element){
 		return this.optional(element) || (value.match(/^\d+$/g) && value >= 0 && value <= 65535);
 	}, "Please enter a port number less than 65536.");
@@ -119,7 +142,7 @@
 		onkeyup: false,
 		groups:{
 			server_ipv4: "server_ip_address_1 server_ip_address_2 server_ip_address_3 server_ip_address_4",
-			server_ipv6: "ip_address_1 ip_address_2 ip_address_3 ip_address_4 ip_address_5 ip_address_6 ip_address_7 ip_address_8"
+			server_ipv6: "ip6_address_r1 ip6_address_r2 ip6_address_r3 ip6_address_r4 ip6_address_r5 ip6_address_r6 ip6_address_r7 ip6_address_r8"
 		},
         rules: {           
 			start_port: {
@@ -149,37 +172,37 @@
             }
 			,server_ip_address_4: {
                 required: isIp4AddrRequired,
-				ip4: true
+				ip4_end: true
             }
-            ,ip_address_1:{
+            ,ip6_address_r1:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             } 
-            ,ip_address_2:{
+            ,ip6_address_r2:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }  
-            ,ip_address_3:{
+            ,ip6_address_r3:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }  
-            ,ip_address_4:{
+            ,ip6_address_r4:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }  
-            ,ip_address_5:{
+            ,ip6_address_r5:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }  
-            ,ip_address_6:{
+            ,ip6_address_r6:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }   
-            ,ip_address_7:{
+            ,ip6_address_r7:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }  
-            ,ip_address_8:{
+            ,ip6_address_r8:{
             	required: isIp6AddrRequired,
             	hexadecimal: true            	
             }            
@@ -223,35 +246,55 @@
 	    var host2 = parseInt($("#server_ip_address_3").val());
 	    var host3 = parseInt($("#server_ip_address_4").val());
 
- 		if (IsBlank("ip6_address_r") && (ip.length > 3)) {
-		    if (ip == jsGatewayIP){
-				jAlert("Server IP can't be equal to the Gateway IP address !");
+ 		if (IsBlank("server_ip_address_") && IsBlank("ip6_address_r")) {
+	   	  	jAlert("Please input valid server address !");
 				return;
 			}
 
-			//alert(jsNetMask);
-			if(jsNetMask.indexOf('255.255.255') >= 0){
-				//the first three field should be equal to gw ip field
-				if((jsGwIP[0] != host0) || (jsGwIP[1] != host1) || (jsGwIP[2] != host2)){
-				  var msg = 'Server IP addr is not in valid range:\n' + jsGwIP[0]+'.'+jsGwIP[1]+'.'+jsGwIP[2]+'.[2~254]';
-				  jAlert(msg);
-				  //jAlert('Host IP is not in valid range:\n' + jsGwIP[0]+'.'+jsGwIP[1]+'.'+jsGwIP[2]+'.[2~254]');
-				  return;
+		if (!IsBlank("server_ip_address_")) {
+			//to check if "Server IPv4 Address" is in "DHCP Pool range"
+			var IPv4_not_valid = false;
+			for(i=0;i<4;i++){
+				j = i+1;
+				if(parseInt(beginArr[i]) > parseInt($("#server_ip_address_"+j).val()) || parseInt($("#server_ip_address_"+j).val()) > parseInt(endArr[i])){
+					IPv4_not_valid = true;
+					break;
 				}		
 			}
-			else if(jsNetMask == "255.255.0.0"){
-				if((jsGwIP[0] != host0) || (jsGwIP[1] != host1)){
-				  jAlert('Server IP addr is not in valid range:\n' + jsGwIP[0]+ '.' + jsGwIP[1] + '.[0~255]' + '.[2~254]');
+
+			//IPv4 validation
+			if (ip == jsGatewayIP){
+				jAlert("Server IP can't be equal to the Gateway IP address !");
+				return;
+			} else if(IPv4_not_valid){
+				jAlert("Server IP addr is not in valid range:\n"+beginAddr+" ~ "+endAddr);
 				  return;
 				}		
 			}
-			else{
-				if(jsGwIP[0] != host0){
-				  jAlert("Server IP addr is not in valid range:\n [10.0.0.2 ~ 10.255.255.254]");
+
+		if (!IsBlank("ip6_address_r")) {
+		//IPv6 validation
+		//Check if IPv6 Mode - Stateless(Auto-Config), Stateful(Use Dhcp Server)
+			if("<?php echo $state; ?>" == "Stateful"){
+				//if Stateful(Use Dhcp Server) then accept inrange values
+				var start = "<?php echo $v6_begin_addr; ?>";
+				var start1 = start.split(":");
+				var end = "<?php echo $v6_end_addr; ?>";
+				var end1 = end.split(":");
+				var ipv6res = ipv6addr.split(":");
+				var ipv6res1 = ipv6res.splice(4, 4);
+
+				for (i = 0; i < ipv6res1.length; i++) {
+					var val = parseInt(ipv6res1[i].toUpperCase(), 16);
+					var low = parseInt(start1[i].toUpperCase(), 16);
+					var upp = parseInt(end1[i].toUpperCase(), 16);
+					if(!((val >= low) && (val <= upp))){
+						jAlert("Server IPv6 addr is not in valid range:\n <?php echo $prefix_arr[0].':'.$v6_begin_addr.' ~ '.$prefix_arr[0].':'.$v6_end_addr; ?>");
 				  return;
 				}		
 			}
 		}
+	     	}
 
 		$('.port').each(function(){
 			if (!validator.element($(this))){
@@ -260,14 +303,10 @@
 			}
 		});
 
-		if (IsBlank("server_ip_address_") && IsBlank("ip6_address_r")) {
-	   	  jAlert("Please input valid server address !");
-	   	  return;
-	    }
-	    else if (IsBlank("server_ip_address_")) {
+		if (IsBlank("server_ip_address_")) {
 	   	    ip = "255.255.255.255";
 	    }
-	    else if (IsBlank("ip6_address_r")) {
+		if (IsBlank("ip6_address_r")) {
 		    ipv6addr = "x"; 
 	    } 
 
@@ -295,13 +334,36 @@
 	}); //end of save btn click
 
 //=================================================
+
+service_names = ['FTP', 'AIM', 'HTTP', 'PPTP', 'HTTPs', 'Telnet', 'SSH'];
+if(service_names.indexOf("<?php echo $service_name; ?>") < 0){
+	var service_name='<?php echo $service_name; ?>';
+	var startport	='<?php echo $startport; ?>';
+	var endport	='<?php echo $endport; ?>';
+} else {
+	var service_name='';
+	var startport	='';
+	var endport	='';
+}
+
+$("#service_name, #start_port, #end_port").change(function() {
+	if(service_names.indexOf($("#service_name").val()) < 0){
+		service_name=$("#service_name").val();
+		startport	=$("#start_port").val();
+		endport	=$("#end_port").val();
+	}
+});
+
 function update_service_field() {
 	var $common_select = $("#common_services");
     var $other = $("#service_name");
 
     if($common_select.find("option:selected").val() == "other") {
         $other.prop("disabled", false).removeClass("disabled").closest(".form-row").show();
-        $("#start_port, #end_port").prop("disabled", false); // Reset ports for user entered numbers
+        $("#start_port, #end_port").prop("disabled", false);
+	$other.val(service_name);
+	$("#start_port").val(startport);
+	$("#end_port").val(endport);
     } 
     else {
         $other.prop("disabled", true).removeClass("disabled").val("").closest(".form-row").hide();
@@ -320,7 +382,7 @@
  // Monitor Common Services because it informs value and visibility of other field
     $("#common_services").change(function() {
         update_service_field();
-    }).trigger("change");
+    });
 
 	var ipv6_arr = populateIPv6Addr(jsV6ServerIP === 'x' ? '' : jsV6ServerIP);
 
@@ -333,6 +395,12 @@
     $("#ip6_address_r7").val(ipv6_arr[6]);
     $("#ip6_address_r8").val(ipv6_arr[7]);
 	
+	if(DeviceMode == "Ipv4"){
+		$("#ip6_address_r5, #ip6_address_r6, #ip6_address_r7, #ip6_address_r8").prop("disabled", true);
+    	} else {
+		$("#ip6_address_r5, #ip6_address_r6, #ip6_address_r7, #ip6_address_r8").prop("disabled", false);
+	}
+	
 });
 </script>
 
@@ -342,13 +410,14 @@
 
     <div  id="educational-tip">
         <p class="tip"> Edit a rule for port forwarding services by user.</p>
-        <p class="hidden">Some more text to help the customer understand about this content.</p>
+        <p class="hidden">Port forwarding permits communications from external hosts by forwarding them to a particular port.</p>
+		<p class="hidden">Port forwarding settings can affect the Gateway's performance.</p>
     </div>
 
 	<form method="post" id="pageForm" action="">
 	<div  class="module forms">
 		<h2>Edit Port Forward</h2>
-		<?php $serviceArr = array('FTP', 'AIM', 'HTTP', 'PPTP'); ?>
+		<?php $serviceArr = array('FTP', 'AIM', 'HTTP', 'PPTP', 'HTTPs', 'Telnet', 'SSH'); ?>
 
 		<div  class="form-row odd">
 					<label for="common_services">Common Service:</label>
@@ -357,6 +426,9 @@
 					<option <?php if(!strcasecmp($service_name, 'AIM')) echo 'selected'; ?> value="5190|5190">AIM</option>
 					<option <?php if(!strcasecmp($service_name, 'HTTP')) echo 'selected'; ?> value="80|80" >HTTP</option>
 					<option <?php if(!strcasecmp($service_name, 'PPTP')) echo 'selected'; ?> value="1723|1723">PPTP</option>
+					<option <?php if(!strcasecmp($service_name, 'HTTPs')) echo 'selected'; ?> value="443|443">HTTPs</option>
+					<option <?php if(!strcasecmp($service_name, 'Telnet')) echo 'selected'; ?> value="23|23">Telnet</option>
+					<option <?php if(!strcasecmp($service_name, 'SSH')) echo 'selected'; ?> value="22|22">SSH</option>
 					<option <?php if(!in_array($service_name, $serviceArr)) echo 'selected'; ?> value="other" class="other">Other</option>
 					</select>
 				</div>
@@ -386,25 +458,25 @@
 		}
 		?>
 
-			<label for="server_ip_address_1">Server IP Address:</label>
-		    <input type="text" size="2" id="server_ip_address_1" value="<?php echo $ip[0];?>" name="server_ip_address_1" class="ipv4-addr smallInput" />
+			<label for="server_ip_address_1">Server IPv4 Address:</label>
+		    <input type="text" size="2" maxlength="3" id="server_ip_address_1" value="<?php echo $ip[0];?>" name="server_ip_address_1" class="ipv4-addr smallInput" />
 	        <label for="server_ip_address_2" class="acs-hide"></label>
-		   .<input type="text" size="2" id="server_ip_address_2" value="<?php echo $ip[1];?>" name="server_ip_address_2" class="ipv4-addr smallInput" />
+		   .<input type="text" size="2" maxlength="3" id="server_ip_address_2" value="<?php echo $ip[1];?>" name="server_ip_address_2" class="ipv4-addr smallInput" />
 	        <label for="server_ip_address_3" class="acs-hide"></label>
-		   .<input type="text" size="2" id="server_ip_address_3" value="<?php echo $ip[2];?>" name="server_ip_address_3" class="ipv4-addr smallInput" />
+		   .<input type="text" size="2" maxlength="3" id="server_ip_address_3" value="<?php echo $ip[2];?>" name="server_ip_address_3" class="ipv4-addr smallInput" />
 	        <label for="server_ip_address_4" class="acs-hide"></label>
-		   .<input type="text" size="2" id="server_ip_address_4" value="<?php echo $ip[3];?>" name="server_ip_address_4" class="ipv4-addr smallInput" />
+		   .<input type="text" size="2" maxlength="3" id="server_ip_address_4" value="<?php echo $ip[3];?>" name="server_ip_address_4" class="ipv4-addr smallInput" />
 		</div>
 
 		<div class="form-row odd">		
 			<label for="ip6_address_r1">Server IPv6 Address:</label>
-			<input type="text" size="1" maxlength="4" id="ip6_address_r1" name="ip_address_1" class="ipv6-addr ipv6-input"/>:
+			<input type="text" size="1" maxlength="4" id="ip6_address_r1" name="ip_address_1" disabled="disabled" class="ipv6-addr ipv6-input"/>:
 	        <label for="ip6_address_r2" class="acs-hide"></label>
-			<input type="text" size="1" maxlength="4" id="ip6_address_r2" name="ip_address_2" class="ipv6-addr ipv6-input"/>:
+			<input type="text" size="1" maxlength="4" id="ip6_address_r2" name="ip_address_2" disabled="disabled" class="ipv6-addr ipv6-input"/>:
 	        <label for="ip6_address_r3" class="acs-hide"></label>
-			<input type="text" size="1" maxlength="4" id="ip6_address_r3" name="ip_address_3" class="ipv6-addr ipv6-input"/>:
+			<input type="text" size="1" maxlength="4" id="ip6_address_r3" name="ip_address_3" disabled="disabled" class="ipv6-addr ipv6-input"/>:
 	        <label for="ip6_address_r4" class="acs-hide"></label>
-			<input type="text" size="1" maxlength="4" id="ip6_address_r4" name="ip_address_4" class="ipv6-addr ipv6-input"/>:
+			<input type="text" size="1" maxlength="4" id="ip6_address_r4" name="ip_address_4" disabled="disabled" class="ipv6-addr ipv6-input"/>:
 	        <label for="ip6_address_r5" class="acs-hide"></label>
 			<input type="text" size="1" maxlength="4" id="ip6_address_r5" name="ip_address_5" class="ipv6-addr ipv6-input"/>:
 	        <label for="ip6_address_r6" class="acs-hide"></label>
@@ -415,10 +487,6 @@
 			<input type="text" size="1" maxlength="4" id="ip6_address_r8" name="ip_address_8" class="ipv6-addr ipv6-input"/>
     	</div>
 
-    	<?php 
-    	$startport = getStr("Device.NAT.PortMapping.".$i.".ExternalPort"); 
-    	$endport   = getStr("Device.NAT.PortMapping.".$i.".ExternalPortEndRange");
-    	?>
 		<div  class="form-row ">
 			<label for="start_port">Start Port:</label>  
 			<input type="text" class="port" value="<?php echo $startport; ?>" id="start_port" name="start_port" />
