--- ./webui/source/Styles/xb3/code/dynamic_dns_add.php
+++ ./webui/source/Styles/xb3/code/dynamic_dns_add.php	2015-04-22 14:22:31.206843009 -0600
@@ -152,12 +152,18 @@
 
    		<div class="form-row " id="Service_Provider" >
 
-   			<div class="form-row "  style="float:left">
    				<label for="Service_Provider">Service Provider:</label>
-   				<input type="text"  id="Service_Provider1"  name="Service_Provider" />
-   			</div>
-   			<div  style="font-size: .8em;  "></br>Eg:dyndns.org,changeip.com,freedns.afraid.org</div>
-
+				<!--input type="text" value="dyndns.org" id="Service_Provider1"  name="Service_Provider" /-->
+				<select name="Service_Provider1" id="Service_Provider1">
+				<?php
+					$ids=explode(",",getInstanceIDs("Device.X_CISCO_COM_DDNS.Service."));
+					foreach ($ids as $key=>$j) {
+						$spName		=getStr("Device.X_CISCO_COM_DDNS.Service.".$j.".ServiceName");
+						$serviceStatus	=getStr("Device.X_CISCO_COM_DDNS.Service.".$j.".Enable");
+						if(strcasecmp($serviceStatus,"false") == 0) echo '<option value="'.$spName.'">'.$spName.'</option>';
+					}
+				?>
+				</select>
    		</div>
 
    		<div class="form-row odd" >
