--- ./webui/source/Styles/xb3/code/firewall_settings_ipv4.php
+++ ./webui/source/Styles/xb3/code/firewall_settings_ipv4.php	2015-04-22 14:22:31.206843009 -0600
@@ -45,11 +45,11 @@
 	
 	$("#disable_firewall").change(function(){
 		if($("#disable_firewall").prop("checked")) {
-			$("#block_http").prop("disabled",true);
-			$("#block_icmp").prop("disabled",true);
-			$("#block_multicast").prop("disabled",true);
-			$("#block_peer").prop("disabled",true);
-			$("#block_ident").prop("disabled",true);
+			$("#block_http").prop("disabled",true).attr('checked', false);
+			$("#block_icmp").prop("disabled",true).attr('checked', false);
+			$("#block_multicast").prop("disabled",true).attr('checked', false);
+			$("#block_peer").prop("disabled",true).attr('checked', false);
+			$("#block_ident").prop("disabled",true).attr('checked', false);
 		}
 		else {
 			$("#block_http").prop("disabled",false);
