--- ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_wireless_network_configuration.php
+++ ./webui/source/Styles/xb3/code/actionHandler/ajaxSet_wireless_network_configuration.php	2015-06-10 14:05:49.101030780 -0600
@@ -26,12 +26,19 @@
 	else if ("save_advance" == $arConfig['sub_target'])
 	{
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_CTSProtectionMode", $arConfig['BG_protect_mode'], false);
+		setStr("Device.WiFi.Radio.$i.X_COMCAST_COM_IGMPSnoopingEnable", $arConfig['IGMP_Snooping'], false);
 		setStr("Device.WiFi.Radio.$i.OperatingChannelBandwidth", $arConfig['channel_bandwidth'], false);
 		setStr("Device.WiFi.Radio.$i.GuardInterval", $arConfig['guard_interval'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_ReverseDirectionGrant", $arConfig['reverse_enabled'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_AggregationMSDU", $arConfig['MSDU_enabled'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_AutoBlockAck", $arConfig['blockACK_enabled'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_DeclineBARequest", $arConfig['blockBA_enabled'], false);
+		
+		//DFS_Support1 1-supported 0-not supported
+		if (("2" == $i) && (getStr("Device.WiFi.Radio.$i.X_COMCAST_COM_DFSSupport") == 1)){
+			setStr("Device.WiFi.Radio.$i.X_COMCAST_COM_DFSEnable", $arConfig['DFS_Selection'], false);
+		}
+		
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_HTTxStream", $arConfig['HT_TxStream'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_HTRxStream", $arConfig['HT_RxStream'], false);
 		setStr("Device.WiFi.Radio.$i.X_CISCO_COM_STBCEnable", $arConfig['STBC_enabled'], false);
@@ -129,11 +136,13 @@
 	
 	if ("PushButton" == $arConfig['pair_method']) 
 	{
-		setStr("Device.WiFi.AccessPoint.$i.WPS.X_CISCO_COM_ActivatePushButton", "true", true);
+		setStr("Device.WiFi.AccessPoint.1.WPS.X_CISCO_COM_ActivatePushButton", "true", true);
+		setStr("Device.WiFi.AccessPoint.2.WPS.X_CISCO_COM_ActivatePushButton", "true", true);
 	}
 	else 
 	{
-		setStr("Device.WiFi.AccessPoint.$i.WPS.X_CISCO_COM_ClientPin", $arConfig['pin_number'], true);
+		setStr("Device.WiFi.AccessPoint.1.WPS.X_CISCO_COM_ClientPin", $arConfig['pin_number'], true);
+		setStr("Device.WiFi.AccessPoint.2.WPS.X_CISCO_COM_ClientPin", $arConfig['pin_number'], true);
 	}
 	
 	// for ($j=0; $j<16; $j++)
@@ -152,7 +161,8 @@
 }
 else if ("pair_cancel" == $arConfig['target'])
 {
-	setStr("Device.WiFi.AccessPoint.$i.WPS.X_CISCO_COM_CancelSession", "true", true);
+	setStr("Device.WiFi.AccessPoint.1.WPS.X_CISCO_COM_CancelSession", "true", true);
+	setStr("Device.WiFi.AccessPoint.2.WPS.X_CISCO_COM_CancelSession", "true", true);
 	echo $jsConfig;	
 }
 else if ("mac_ssid" == $arConfig['target'])
@@ -181,16 +191,18 @@
 	}
 	
 	$at = array();
-	
-	if ("5"==$i)		//HotSpot clients do not exist in Host Table.
-	{
-		$ssids = explode(",", getInstanceIds("Device.X_COMCAST_COM_GRE.SSID."));
-		foreach($ssids as $id)
-		{
-			$clients = explode(",", getInstanceIds("Device.X_COMCAST_COM_GRE.SSID.$id.AssociatedDevice."));
+	//HotSpot clients do not exist in Host Table.
+	//Device.X_COMCAST_COM_GRE.SSID.1. is for SSID-5
+	//Device.X_COMCAST_COM_GRE.SSID.2. is for SSID-6
+	if ("5"==$i || "6"==$i)
+	{
+		$id = ("5"==$i)?"1":"2";
+		$clients = explode(",", getInstanceIds("Device.X_COMCAST-COM_GRE.Tunnel.1.SSID.$id.AssociatedDevice."));
+		//explode on empty string returns array count as 1 [with string(0) ""]
+		if($clients[0]){
 			foreach($clients as $v)
 			{
-				array_push($at, array(getStr("Device.X_COMCAST_COM_GRE.SSID.$id.AssociatedDevice.$v.Hostname"), getStr("Device.X_COMCAST_COM_GRE.SSID.$id.AssociatedDevice.$v.MACAddress")));
+				array_push($at, array(getStr("Device.X_COMCAST-COM_GRE.Tunnel.1.SSID.$id.AssociatedDevice.$v.Hostname"), getStr("Device.X_COMCAST-COM_GRE.Tunnel.1.SSID.$id.AssociatedDevice.$v.MACAddress")));
 			}		
 		}
 	}
@@ -220,7 +232,9 @@
 else if ("save_filter" == $arConfig['target'])
 {
 	$ssids = array($i);
-	if ("5"==$i){
+
+	//xfinitywifi[HotSpot] filter rule apply to both 5 & 6 SSID
+	if ("5"==$i || "6"==$i){
 		$ssids = array("5","6");
 	}
 	
@@ -296,6 +310,19 @@
 			$filter_block  = "false";
 		}
 		
+		$get_filter_enable = getStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.Enable");
+		$get_filter_block  = getStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.FilterAsBlackList");
+
+		/*------When changing from "allow_all" to "allow" go from "allow_all" to "deny" then to "allow" -----*/
+		if(($get_filter_enable == "false" && $get_filter_block == "false") && ($filter_enable == "true" && $filter_block == "false")){
+			//"allow_all" to "deny"
+			setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.Enable", "true", false);
+			setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.FilterAsBlackList", "true", true);
+			//"deny" to "allow"
+			setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.Enable", "true", false);
+			setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.FilterAsBlackList", "false", true);
+		}
+
 		setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.Enable", $filter_enable, false);
 		setStr("Device.WiFi.AccessPoint.$i.X_CISCO_COM_MACFilter.FilterAsBlackList", $filter_block, true);	
 		
